spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -148,6 +149,13 @@ object DistriOptimizer {
     var dropModelNumBatch = 0
     var lossArray = new Array[Double](_subModelNumber)
 
+    // gradient clip settings
+    val constantClippingEnable = clippingParams.enableConstantClipping
+    val normClippingEnable = clippingParams.enableL2NormClipping
```

Comment:
what if both l2norm and constant clipping are set to true?

Reply:
It will use l2norm first and then constant clipping

Reply:
Does it make sense?

Reply:
or shall we make it a single interface and the method of clipping as parameters? e.g. interface setGradClip(['const']) or setGradClip(['l2']) if I want one of the two, and if I want to stack those I shall write setGradClip(['const', 'l2']), the order also matters.